nixos/modules/services/databases/influxdb2.nix

```diff
@@ -0,0 +1,52 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+  cfg = config.services.influxdb2;
+  configFile = pkgs.writeText "config.json" (builtins.toJSON cfg.settings);
+in
+{
+  options = {
+    services.influxdb2 = {
+      enable = mkEnableOption "the influxdb2 server";
+      package = mkOption {
+        default = pkgs.influxdb2;
+        defaultText = "pkgs.influxdb2";
+        description = "influxdb2 derivation to use";
+        type = types.package;
+      };
+      settings = mkOption {
+        default = { };
+        description = "configuration options for influxdb2";
+        type = types.attrs;
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    assertions = [{
+      assertion = !(builtins.hasAttr "bolt-path" cfg.settings) && !(builtins.hasAttr "engine-path" cfg.settings);
+      message = "services.influxdb2.config: bolt-path and engine-path should not be set as they are managed by systemd";
+    }];
+    systemd.services.influxdb2 = {
```

Comment:
Can you comment on the differences between your service and the [upstream](https://github.com/influxdata/influxdb/blob/master/scripts/influxdb.service) service? I'm noticing upstream uses `Type=forking`.

Reply:
Missed that one.

Reply:
The main difference is the use of dynamic users.

Reply:
Sounds good. Thanks!